### PR TITLE
Feature - 5692 - Update Application Card Styles

### LINKED ISSUE #5692: ✨ Update application "cards" on "My applications" to the future dashboard specs
## ✨ Feature

Update application "cards" on "My applications" to the future dashboard specs.

## 🕵️ Details

The dashboard work by @substrae offers a new way to present applications that a user has submitted. This issue is to rebuild the "card" that we use to present them to users.

The "get support" link should be the email associated with the "pool".
It's ok if archiving functionality goes away. We will be replacing it shortly.

## 🎨 Design File

(see epic)

## 📸 Screenshot

New Draft application card:

New submitted application card:

This is what the cards currently look like:

## ✅ Acceptance Criteria

A set of assumptions which, when tested, verify that the feature was properly implemented.

Draft Card 
- [ ] Apply By - should show pool expiry date
- [ ] Draft applications show a "Delete this application" link that, when clicked, triggers a dialog asking the user to confirm their decision to delete the application
- [ ] Continue this application - should open the application page in edit mode 

Submitted Card and Application History card
- [ ] Applied On  - show show the application submitted date
- [ ] View my application link -  when clicked, takes the user to the summary of their application
- [ ] Application status should show inside the pill 

All Cards 
- [ ] Clicking on an application's "See job ad" link takes the user to the appropriate job poster page
- [ ] Clicking on an application's "Get support" link triggers the user's configured email client with the relevant recruitment administrator's email address populated
- [ ]  If there is no data on either of the status the cards don't appear and the page remains as it is right now 
Screen Shot 2023-02-13 at 10 42 44 AM

### PR DESCRIPTION
**Pull request by esizer**
🤖 Resolves #5692 

## 👋 Introduction

Updates the `<ApplicationCard />` to match the [new mockups](https://www.figma.com/file/jHsXzqMfBtomLa1Gom94Y7/Profile-dashboard?node-id=310%3A4422&t=sIwuEnRA2Fx56esm-0).

## 🕵️ Details

~~I omitted the ID, not sure if this was the right call. They were taking up a lot of space and I'm not sure how much value they provide 🤔~~

ID added back in on [the advice](https://github.com/GCTC-NTGC/gc-digital-talent/pull/5744#issuecomment-1435166490) from @Jerryescandon 

## 📋 TO DO

- [x] Add translations

## 🧪 Testing

1. Run storybook `npm run storybook`
2. Navigate to the "Application Cards" story
3. Confirm the styles match [new mockups](https://www.figma.com/file/jHsXzqMfBtomLa1Gom94Y7/Profile-dashboard?node-id=310%3A4422&t=sIwuEnRA2Fx56esm-0)
4. Confirm the proper action links appear depending on statues and expiry.

## 📸 Screenshot

Screenshot 2023-02-17 142232

### CODE REVIEW

**Comment by vd1992** [comment]
@Jerryescandon 
any strong feelings about the below? 

> ## 🕵️ Details
> 
> I omitted the ID, not sure if this was the right call. They were taking up a lot of space and I'm not sure how much value they provide 🤔

**Comment by Jerryescandon** [comment]
Well, the purpose of IDs is troubleshoot, so unfortunately there's no easy way of tracking usefulness cause they dont come up until it's a problem. It's my instinct that this is valuable information that we should display somewhere and I would vote to keep this in. I think it's @gobyrne 's call though.

**Comment by vd1992** [comment]
I don't think this is a consequence of this PR, but just marking it as an issue, it appears editing an experience within the application context doesn't maintain the context and kicks you out of it
might be a side-effect of #5683 

https://recordit.co/v2FhLxABww

File: apps/web/src/pages/Applications/MyApplicationsPage/components/ApplicationCard/ApplicationCard.tsx [hunk 1]
```diff
@@ -41,92 +45,113 @@ const ApplicationCard = ({
...
-  let borderKey: BorderMapKey = "dt-gray";
-  if (!application.archivedAt && application.status) {
-    borderKey = borderKeyMap[application.status];
-  }
-
-  const border = borderMap[borderKey];
+  const recruitmentIsExpired = isExpired(
+    application.poolAdvertisement?.closingDate,
+  );
+  const submittedAt = formatSubmittedAt(application.submittedAt, intl);
```
Comment (vd1992) [inline apps/web/src/pages/Applications/MyApplicationsPage/components/ApplicationCard/ApplicationCard.tsx:53]: submittedAt is not passed into this, it is missing from the graphql operation
https://github.com/GCTC-NTGC/gc-digital-talent/blob/4e0cfeb633af5624ac6deb28621d28b21679aca5/apps/web/src/pages/Applications/MyApplicationsPage/applicationOperations.graphql#L1-L39

resulting in 
image

File: apps/web/src/pages/Applications/MyApplicationsPage/components/ApplicationCard/ApplicationActions.tsx [hunk 2]
```diff
@@ -72,6 +115,30 @@ const SeeAdvertisementAction = ({
...
+      {intl.formatMessage({
+        defaultMessage: "Get support",
+        id: "rXdaZW",
+        description: "Link text to direct a user to the support page",
+      })}
+    </Link>
+  );
+};
 
 export interface DeleteActionProps extends ActionProps {
```
Comment (vd1992) [inline apps/web/src/pages/Applications/MyApplicationsPage/components/ApplicationCard/ApplicationActions.tsx:143]: opening delete dialog throws this
might have been an existing issue?

image

File: apps/web/src/pages/Applications/MyApplicationsPage/components/ApplicationCard/ApplicationActions.tsx [hunk 3]
```diff
@@ -92,11 +159,12 @@ const DeleteAction = ({ show, application, onDelete }: DeleteActionProps) => {
   return (
```
Comment (vd1992) [inline apps/web/src/pages/Applications/MyApplicationsPage/components/ApplicationCard/ApplicationActions.tsx:159]: same for archive action

image

**Review by vd1992 — CHANGES_REQUESTED** [review]

**Comment by vd1992** [comment]
question

is reversing archiving desired?

image

is application status intentionally omitted?

image

**Comment by esizer** [comment]
> is reversing archiving desired?

This has never been a function and imo out of scope for a simple restyle of the card.

> is application status intentionally omitted?

The [new mockups](https://www.figma.com/file/jHsXzqMfBtomLa1Gom94Y7/Profile-dashboard?node-id=310%3A4422&t=sIwuEnRA2Fx56esm-0) do not include statuses, no.

File: apps/web/src/pages/Applications/MyApplicationsPage/components/ApplicationCard/ApplicationCard.tsx (hunk 1 above)
Comment (esizer) [reply apps/web/src/pages/Applications/MyApplicationsPage/components/ApplicationCard/ApplicationCard.tsx:53]: Good catch! Added in [add submittedAt to application query](https://github.com/GCTC-NTGC/gc-digital-talent/pull/5744/commits/251c4c02883c09a759aeeb39a2bff11c3d8796c2)

File: apps/web/src/pages/Applications/MyApplicationsPage/components/ApplicationCard/ApplicationActions.tsx (hunk 2 above)
Comment (esizer) [reply apps/web/src/pages/Applications/MyApplicationsPage/components/ApplicationCard/ApplicationActions.tsx:143]: Definitely a preexisting issue. Fixed in [fix issue with nested p tags in application dialogs](https://github.com/GCTC-NTGC/gc-digital-talent/pull/5744/commits/42f9289e919319fd6e0623f98b52f3802edfafec)

File: apps/web/src/pages/Applications/MyApplicationsPage/components/ApplicationCard/ApplicationActions.tsx (hunk 3 above)
Comment (esizer) [reply apps/web/src/pages/Applications/MyApplicationsPage/components/ApplicationCard/ApplicationActions.tsx:159]: Also fixed in [fix issue with nested p tags in application dialogs](https://github.com/GCTC-NTGC/gc-digital-talent/pull/5744/commits/42f9289e919319fd6e0623f98b52f3802edfafec)

**Comment by vd1992** [comment]
> This has never been a function and imo out of scope for a simple restyle of the card.

yeah, I don't really think it'd be of much use anyway

> The [new mockups](https://www.figma.com/file/jHsXzqMfBtomLa1Gom94Y7/Profile-dashboard?node-id=310%3A4422&t=sIwuEnRA2Fx56esm-0) do not include statuses, no.

ah, thanks

**Comment by vd1992** [comment]
this story should probably be reconsidered now that statuses have no effect on the card

https://61e099a1bb1465003a73d3ce-ejsoifbflj.chromatic.com/?path=/story/components-application-card--application-cards

**Comment by vd1992** [comment]
> The statuses do have an effect on the card, they just aren't displayed. I still listed them because it affects which links are displayed in the bottom of the card using the [utility functions](https://github.com/GCTC-NTGC/gc-digital-talent/blob/42f9289e919319fd6e0623f98b52f3802edfafec/apps/web/src/pages/Applications/MyApplicationsPage/components/ApplicationCard/utils.ts#L12-L50).

ah yes, correct


should probably get rid of the "continue" for ones where the pool is closed

image

**Comment by esizer** [comment]
> should probably get rid of the "continue" for ones where the pool is closed

Which status is this?

File: apps/web/src/pages/Applications/MyApplicationsPage/components/ApplicationCard/ApplicationCard.tsx [hunk 4]
```diff
@@ -41,92 +45,113 @@ const ApplicationCard = ({
...
+          <ApplicationActions.ArchiveAction
+            onArchive={onArchive}
+            show={applicationCanBeArchived}
+            application={application}
+          />
         </div>
+        <ApplicationActions.ContinueAction
+          show={applicationIsDraft}
+          application={application}
+        />
```
Comment (vd1992) [inline apps/web/src/pages/Applications/MyApplicationsPage/components/ApplicationCard/ApplicationCard.tsx:154]: I think it should be like
`show={applicationIsDraft && !recruitmentIsExpired}`

Comment (esizer) [reply apps/web/src/pages/Applications/MyApplicationsPage/components/ApplicationCard/ApplicationCard.tsx]: Thanks, added in [hide continue application if expired](https://github.com/GCTC-NTGC/gc-digital-talent/pull/5744/commits/62b159ae85e113a05c244a3c742071c1fee8f6f6)

**Comment by vd1992** [comment]
looking at the `untranslated.json` file
some descriptions will need to be updated as they are accidental copy-paste mistakes I think?

image

**Comment by esizer** [comment]
> looking at the `untranslated.json` file
> some descriptions will need to be updated as they are accidental copy-paste mistakes I think?

Oh, thanks! Heh, some of these may be old errors because I just touched the `defaultMessage` 😅 

Anyway, fixed in [update descriptions for application card strings](https://github.com/GCTC-NTGC/gc-digital-talent/pull/5744/commits/a8035d071e69e89f4274dd92fc8d7b8b167aa9a5)

**Review by vd1992 — APPROVED** [review]
I've run out of things to check, so, all good on my end.
Great!

**Comment by brindasasi** [comment]
> 3\. new mockups
oops .. sorry I just saw this. Cards should show the status inside the pill  and its in the A/C as well 
Screen Shot 2023-02-21 at 5 19 55 PM

**Review by brindasasi — DISMISSED** [review]
> image

Application status should be shown inside the pill like below
Screen Shot 2023-02-21 at 5 28 41 PM

**Comment by esizer** [comment]
> Application status should be shown inside the pill like below

Aahh, I was not sure if that was something out of scope since "pending skill assessments" is not an existing status. Thanks! 😄 

Added in [add status to application card if not draft](https://github.com/GCTC-NTGC/gc-digital-talent/pull/5744/commits/34298ae308fe002c242be642513c44ad098769a3)

**Comment by brindasasi** [comment]
> > Application status should be shown inside the pill like below
> 
> Aahh, I was not sure if that was something out of scope since "pending skill assessments" is not an existing status. Thanks! 😄
> 
> Added in [add status to application card if not draft](https://github.com/GCTC-NTGC/gc-digital-talent/pull/5744/commits/34298ae308fe002c242be642513c44ad098769a3)

woww.. thats fastest turn around. Thank you! 🥳

few nit picks

- Title should be prefixed with level and followed by a comma  as per the mock up.  ex.  **IT-03, Project Manager** . We don't need the stream 
- Pill BG color looks different from the mock up
Screen Shot 2023-02-21 at 6 17 37 PM

image

**Comment by brindasasi** [comment]
> > * Title should be prefixed with level and followed by a comma  as per the mock up.  ex.  **IT-03, Project Manager** . We don't need the stream
> 
> We have a [standardized function for formatting pool advertisement titles](https://github.com/GCTC-NTGC/gc-digital-talent/blob/5c787a10dc609d51fca60e50aae4d5873497c055/frontend/common/src/helpers/poolUtils.tsx#L101-L116) and this would introduce a new format. @Jerryescandon Is this something we are changing or should we keep the existing format?
> 
> > * Pill BG color looks different from the mock up
> 
> I think there is two things going on here.
> 
> 1. Designers are working on a new pill design and I believe this design uses it so our existing pill does not quite match up yet
> 2. It appears from the screenshot that you are viewing this story with dark mode enabled. Can you confirm? We currently do not fully support dark mode but have it enabled in storybook to assist with the development of it.
> 
> Screenshot 2023-02-22 082137

yes you are right.. mine has dark mode enabled 😅. Light mode looks better 👍
Lets confirm the title .. tagging @substrae also for that.

**Comment by Jerryescandon** [comment]
> We have a [standardized function for formatting pool advertisement titles](https://github.com/GCTC-NTGC/gc-digital-talent/blob/5c787a10dc609d51fca60e50aae4d5873497c055/frontend/common/src/helpers/poolUtils.tsx#L101-L116) and this would introduce a new format. @Jerryescandon Is this something we are changing or should we keep the existing format?

To be honest I'm not married to the old format, but consistency above all else. I am ok with this format, but would rather we open a ticket to change the standard to this one, as oppose to create an exception.

**Review by brindasasi — APPROVED** [review]

**Pull request merged**